haystack/preview/components/preprocessors/text_document_splitter.py

```diff
@@ -0,0 +1,109 @@
+from typing import List, Optional, Dict, Any, Literal, Tuple
+
+from more_itertools import windowed
+
+from haystack.preview import component, Document, default_from_dict, default_to_dict
+
+
+@component
+class TextDocumentSplitter:
+    """
+    Split a text document into a list of text documents with shorter texts.
+    This is useful for splitting documents with long texts that otherwise would not fit into the maximum text length of language models
+    """
+
+    def __init__(
+        self, split_by: Literal["word", "sentence", "passage"] = "word", split_length: int = 200, split_overlap: int = 0
```

Comment:
`passage` --> `paragraph`? I've seen this name raise questions many times.

Reply:
I would keep it for now but have no strong opinion. In "DPR" it's also dense **passage** retrieval. If it comes up again, we can change it anytime.